veniq/dataset_collection/dataflow/collectEM.py

```diff
@@ -0,0 +1,63 @@
+from collections import defaultdict
+from typing import List, Dict
+
+import d6tcollect
+import d6tflow
+# from veniq.dataset_collection.augmentation import InvocationType
+from joblib import Parallel, delayed
+
+from veniq.ast_framework import AST
+from veniq.ast_framework import ASTNodeType, ASTNode
+from veniq.dataset_collection.augmentation import collect_info_about_functions_without_params
+from veniq.dataset_collection.dataflow.preprocess import TaskAggregatorJavaFiles
+from veniq.utils.ast_builder import build_ast
+
+d6tcollect.submit = False
+
+
+@d6tflow.requires({'csv': TaskAggregatorJavaFiles})
+class TaskFindEM(d6tflow.tasks.TaskCache):
+    dir_to_search = d6tflow.Parameter()
+    dir_to_save = d6tflow.Parameter()
+    system_cores_qty = d6tflow.IntParameter()
+
+    def _find_EMs(self, row):
+        result_dict = {}
+        try:
+            ast = AST.build_from_javalang(build_ast(row['original_filename']))
```

Comment:
long method. this whole routine about extracting method declrations could be put into a separate method